spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -108,6 +109,12 @@ public Batch toBatch() {
     return this;
   }
 
+  @Override
+  public MicroBatchStream toMicroBatchStream(String checkpointLocation) {
+    return new SparkMicroBatchStream(
```

Comment:
The stats estimation we use in the scan that relies on the latest snapshot should work for batch and micro-batch, right? I think Structured Streaming does not push predicates currently so we will use the snapshot summary if the underlying table is partitioned?

Reply:
good point @aokolnychyi 
`stats estimation` - is something that i felt - will evolve differently for MicroBatchStreaming vs Batch reads - as Batch reads only rely on the latest snapshot - where as the data returned by MicroBatchStreaming is multiple Snapshots. Given the cognitive overload of the current PR - & the sheer number of things to drive consensus on - I want to take that in a different PR. Added to the **Whats Next** section - of this PR. Once the PR - concludes - I will file those as issues & track. Pl. lemme know if that makes sense.

Reply:
Absolutely! We need to check how Spark uses stats for streaming relations. Maybe, it can actually be fine this way as the scan object is reused between batch and micro-batch.

Reply:
I am resolving the comments - the ones which I felt - where we are able to drive reasonable consensus or tracked in my PR **What next** section - so that - **the ones that need more discussion will pop out**. Pl. definitely reopen or start a new comment - if you feel - the need for more discussion right here.